presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -912,6 +912,60 @@ public void testRow()
         assertQuery("SELECT cast(row(nationkey, regionkey) as row(a bigint, b bigint)) FROM nation");
     }
 
+    @Test
+    public void testDecimalRangeFilters()
+    {
+        // Actual session is for the native query runner.
+        Session actualSession = Session.builder(getSession())
+                .setCatalog("hive")
+                .setCatalogSessionProperty("hive", "parquet_pushdown_filter_enabled", "true")
+                .build();
+
+        // Expected session is for the Java query runner.
+        // It does not support Parquet filter pushdown yet, so we have to explicitly disable it.
+        Session expectedSession = Session.builder(actualSession)
+                .setCatalogSessionProperty("hive", "parquet_pushdown_filter_enabled", "false")
+                .build();
+
+        // Generate temporary table name.
+        String tmpTableName = generateRandomTableName();
+
+        try {
```

Comment:
Is this `try-finally` pattern used for other tests as well?

Reply:
Yes, this pattern is currently used in the tests in this suite and others, however, it is not a common pattern:
- testCatalogWithCacheEnabled
- testCreateUnpartitionedTableAsSelect

Typically, it is used when a special session setup is required and we are creating a table. In this case, we need a new table to test various decimal values which are not provided by default as part of the standard TPC-H setup.